scipy/linalg/tests/test_matfuncs.py

```diff
@@ -437,69 +437,64 @@ def test_gh17918(self):
 
     def test_data_size_preservation_uint_in_float_out(self):
```

Comment:
I think all these tests can be replaced with:
```python3
    @pytest.mark.parametrize("dtype",
        [np.uint8, np.uint16, np.uint32, np.uint64, np.int8, np.int16, np.int32,
         np.int64, np.float16, np.float32, np.float64, np.complex64, np.complex128])
    def test_dtype_preservation(self, dtype):
        A = np.eye(3)
        B, = _ensure_dtype_cdsz(A)
        assert sqrtm(A).dtype == B.dtype
        assert sqrtm(-A).dtype == np.result_type(B.dtype, 1j)
```
and include `float128`/`complex256` if they are supported.

But to show that the simplification above is correct, I'll leave the tests as you had them.